### PR TITLE
proposed initial support for site-wide Difficulty settings

### PR DESCRIPTION
**Pull request by Alys**
Goes along with https://github.com/HabitRPG/habitrpg-shared/pull/360

Note that this is going into the "difficulty" feature branch, so it won't go live any time soon.

Trello card: [SITE DIFFICULTY SETTINGS v1 on HabitRPG | Trello](https://trello.com/c/C7CXSo02/428-site-difficulty-settings-v1)

screen shot 2014-11-15 at 12 27 34 pm

### CODE REVIEW

**Comment by lemoness** [comment]
This looks good!

**Comment by lemoness** [comment]
Are the difficulty settings completed?

**Comment by Alys** [comment]
No, not at all. I have a small amount of code locally for a couple of them which I'll tidy up and add here soon. I'll also merge the latest 'develop' changes into the 'difficulty' branch.

There's a fair bit of work involved if we do all of the requested settings, but I'm thinking we could perhaps do just a few now and then more later.

The only thing this PR did was add basic support for difficulty settings (empty user interface, etc). It was merged into the 'difficulty' branch which means it won't go live now. When we're happy with all the changes on this branch, we'd need to merge it into the 'develop' branch before it can go live.

**Comment by eindoofus** [comment]
I see three large problems with this feature:
1. Too broad: settings appear to affect every task
2. Too specific: overwhelming to the non-technical player
3. Nitpicking: takes away from the baseline of the game and unnecessarily scales core mechanics

As an alternative I have the following idea that could be applied to each task:

In the game the rewards and punishments change depending on the frequency of execution of a task. This system sometimes gets in the way of the anticipated reward or punishment. For example, I believe my reward should increase as I have a longer streak of "20 Minute Jog" versus "Call Parents". Or there may be rewards I want to consistently get rewarded for without change like "Study Programming".

I was thinking that line-graph icons could add some user-friendly customization for the handling of red/yellow/green tasks in the game. For example, if I'd like my "Growth of Reward" to increase after each successive jogs I would select [ / ] , call parents I could select [ \ ](which I believe is the current behavior), and study programming could be [--]. The same idea could be applied to "Growth of Punishment."

**Comment by Alys** [comment]
@eindoofus These settings are not related to task difficulty. They will be things like make the health potion unavailable, allow death to remove off-class items, allow multi-day absences to do more damage than one-day absences. They will be in a specific settings screen that players can ignore if they do not understand them or do not want to use them.

**Comment by eindoofus** [comment]
@Alys I'm seeing numerous sliders that appear to be global at [SITE DIFFICULTY SETTINGS v1](https://trello.com/c/C7CXSo02/428-site-difficulty-settings-v1):
- Global task damage slider
- Death penalty slider
- Baseline stats slider
- Experience gain slider

**Comment by Alys** [comment]
If they are implemented, they will still be optional. These are things that people have asked for, so at least some will want them. Those that do not can ignore them.

**Comment by eindoofus** [comment]
@Alys I guess my primary concern was that it'll complicate the UI and confuse non-technical players, but I suppose if it's hidden under the hood then it's not too big of a deal. And since it's a non-traditional game, it's unrealistic to expect players to play it in a consistent way anyway.

However, I would encourage the slider stat ranges to not be too drastic since it may affect the potential for players to group together as the game grows. I personally wouldn't want to join with other players if I saw them growing at quadruple rate because they set all of their sliders to the best position possible. There is already a risk for that by setting task difficulty, but it'll be much more drastic if these slider stats are too powerful.

Basically, I would recommend any handicap sliders to be as extreme as people want them, but reward sliders should be set with caution due to abuse and future group conflict.
